src/platform/common/net/httpClient.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { IHttpClient } from '../types';
+import { IServiceContainer } from '../../ioc/types';
+import { IWorkspaceService } from '../application/types';
+import { traceVerbose } from '../../logging';
+import * as fetch from 'cross-fetch';
+
+@injectable()
+export class HttpClient implements IHttpClient {
+    public readonly requestOptions: RequestInit;
+    constructor(@inject(IServiceContainer) serviceContainer: IServiceContainer) {
```

Comment:
why inject the entire service container instead of the actual dependencies?

Reply:
It was doing that before, I didn't notice. I'll change it if there's another update.